### PR TITLE
add solution

### PR DESCRIPTION
**Pull request by dimarogkov**
[DEMO LINK](https://dimarogkov.github.io/js_promises_practice_DOM/)

### CODE REVIEW

File: src/scripts/main.js [hunk 1]
```diff
@@ -1 +1,55 @@
...
+      resolve('First promise was resolved');
+    });
+
+    setTimeout(() => reject(new Error('First promise was rejected')), 3000);
+  });
+
+  const secondPromise = new Promise((resolve) => {
+    const onClick = () => {
+      resolve('Second promise was resolved');
+    };
```
Comment (vadiimvooo) [inline src/scripts/main.js:16]:
```suggestion
    const onClick = (e) => {
      e.preventDefault();
      resolve('Second promise was resolved');
    };
```

**Review by vadiimvooo — CHANGES_REQUESTED** [review]
Seems like you haven't add notifications for resolved and rejected promises according to task.
image

File: src/scripts/main.js [hunk 2]
```diff
@@ -1 +1,62 @@
+/* eslint-disable no-console */
 'use strict';
+
+document.addEventListener('DOMContentLoaded', () => {
```
Comment (Viktor-Kost) [inline src/scripts/main.js:4]: I would recommend to move your resolvers into own functions

```
const clickPromise = new Promise(resolve => {
  document.addEventListener('click', () => {
    resolve();
  });
});


const contextMenuPromise = new Promise(resolve => {
  document.addEventListener('contextmenu', () => {
    resolve();
  });
});
```

apply Promise.race to the secondPromise

```
const secondPromise = new Promise((resolve, reject) => {
  Promise.race([clickPromise, contextMenuPromise])
    .then((value) => resolve('Second promise was resolved'));
});
```


apply Promise.all to the thirdPromise

```
const thirdPromise = new Promise(resolve => {
  Promise.all([clickPromise, contextMenuPromise])
    .then((value) => resolve('Third promise was resolved'));
});
```

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]

**Review by loralevitska — APPROVED** [review]
Well done)

**Pull request closed without merging**